src/javascript/app_2/pages/trading/components/elements/language_switcher.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { localize } from '../../../../../_common/localize';
+
+class LanguageSwitcher extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            is_collapsed   : false,
+            active_language: this.props.languages[0],
+        };
+    }
+
+    toggleLanguageList = () => {
+        this.setState({
+            is_collapsed: !this.state.is_collapsed,
+        });
+    }
+
+    switchLanguage = (lang) => {
+        this.setState({
+            active_language: lang,
+        });
+    }
+
+    render() {
+        const language_list_collapsed = {
+            visibility: `${this.state.is_collapsed ? 'visible' : 'hidden'}`,
+            position  : 'absolute',
+            width     : '100%',
+        };
+
+        const language_is_active = (lang) => {
+            if (lang.id === this.state.active_language.id) {
+                return true;
+            }
+            return false;
+        };
```

Comment:
nitpick: move to a member function. Just ` return (lang.id === this.state.active_language.id); ` instead of if else.